### PR TITLE
Add ability to use pm-cpu/pm-gpu with maint-2.0

### PR DESCRIPTION
**Pull request by ndkeen**
Add ability to use pm-cpu/pm-gpu with maint-2.0
Rename machine perlmutter to be either pm-cpu (default) or pm-gpu.
Changes for alvarez machine also included to be consistent with master (alvarez is test machine for pm-cpu) 
In this PR, only updated PE layouts that were related to pm-cpu (or pm-gpu/alvarez) -- we may want to
update all of the PE layouts to be more consistent with current master.
Fixes https://github.com/E3SM-Project/E3SM/issues/5284
[bfb]

### CODE REVIEW

**Comment by ndkeen** [comment]
Who best to review?

File: cime_config/allactive/config_pesall.xml [hunk 1]
```diff
@@ -270,6 +270,125 @@
...
+          <ntasks_rof>-4</ntasks_rof>
+          <ntasks_ice>-4</ntasks_ice>
+          <ntasks_ocn>-4</ntasks_ocn>
+          <ntasks_glc>-1</ntasks_glc>
+          <ntasks_wav>-1</ntasks_wav>
+          <ntasks_cpl>-4</ntasks_cpl>
+        </ntasks>
+      </pes>
+    </mach>
+    <mach name="cori-haswell">
```
Comment (rljacob) [inline cime_config/allactive/config_pesall.xml:291]: Did you mean to bring in these layouts for other machines?

Comment (ndkeen) [reply cime_config/allactive/config_pesall.xml:291]: I was trying to only make updates affecting pm-cpu (and pm-gpu/alvarez), but I actually wasn't sure if it make more sense to bring in "all" new changes into files describing PE layouts as it's easier to maintain.  But then there may be some change that is required elsewhere...

Comment (rljacob) [reply cime_config/allactive/config_pesall.xml:291]: Just mention it in the PR description.

**Review by rljacob — APPROVED** [review]

**Comment by ndkeen** [comment]
I just made a commit to this branch to add a PE layout for the RRM northamerica compset (runs on 8 nodes) and a slight tweak to another common compset in the `e3sm_prod` suite which is now running out-of-box for me on pm-cpu.

**Comment by ndkeen** [comment]
Noting here that after the recent PM maintenance, this branch will now require same changes here:
https://github.com/E3SM-Project/E3SM/pull/5373
which I will do soon.

**Pull request merged**